### PR TITLE
DPL-717: Don't filter for fit to pick

### LINKED ISSUE #740: DPL-717-1: Create a new endpoint for creating entities for Lighthouse deep-well plates in Sequencescape
**User story**
As a lab technician, I want to be able to scan in barcodes of deep-well plates received from Lighthouse labs so that the LIMS will make these plates pickable for stamping.

**Who are the primary contacts for this story**
Stuart M, Andrew S, Adrianne L, Ewan H

**Who is the nominated tester for UAT**
TBC

**Acceptance criteria**
To be considered successful the solution must allow:
- [ ] Create an endpoint on Lighthouse Service which allows the submission of a number of plate barcodes for deep-well plates that are to be created in Sequencescape.
- [ ] The endpoint must first check that each plate is recognised as coming in via Lighthouse labs.  i.e. it can be found in MongoDB; if not generate an informative error about this problem.
- [ ] The endpoint will then send a request to Sequencescape to create the plate there.  If the plate already existed, this should be recognised and an informative error generated about the problem.
- [ ] The response from the endpoint will be a summary of the actions taken or the errors generated for each plate.

**References**
This story has a non-blocking relationship with:
- https://github.com/sanger/General-Backlog-Items/issues/283 -- this is an implementation of option 4 on this research story.

**Additional context**

### PR DESCRIPTION
**Pull request by sdjmchattie**
Closes #740 

Changes proposed in this pull request:

* Set whether fit to pick should be filtered for or not per plate type.
* Create methods that pick all samples for a plate when fit to pick filter is set to false.

### CODE REVIEW

File: lighthouse/routes/common/plates.py [hunk 1]
```diff
@@ -43,34 +45,19 @@ def create_plate_from_barcode() -> FlaskResponse:
...
-        # get samples for barcode
-        (fit_to_pick_samples, count_fit_to_pick_samples, _, _, _) = get_fit_to_pick_samples_and_counts(barcode)
-
-        if not fit_to_pick_samples:
-            return bad_request(f"No fit to pick samples for this barcode: {barcode}")
-
-        body = create_post_body(barcode, plate_type, fit_to_pick_samples)
+    plate_configs = app.config["SS_PLATE_CONFIG"]
+    if plate_type not in plate_configs:
+        return bad_request(f"POST request 'type' must be from the list: {', '.join(plate_configs)}")
```
Comment (emrojo) [inline lighthouse/routes/common/plates.py:53]: plate_configs is a dict, not a list

File: lighthouse/helpers/mongo.py [hunk 2]
```diff
@@ -67,6 +67,27 @@ def get_positive_samples_in_source_plate(source_plate_uuid: str) -> Optional[Lis
...
+
+    Arguments:
+        source_plate_uuid {str} -- The source plate UUID for which to get positive samples.
+
+    Returns:
+        {SampleDocs} -- A list of all samples on the source plate; otherwise None if they cannot be determined.
+    """
+    try:
+        samples_collection: Collection = cast(Eve, app).data.driver.db.samples
+        query = {FIELD_LH_SOURCE_PLATE_UUID: source_plate_uuid}
```
Comment (emrojo) [inline lighthouse/helpers/mongo.py:81]: maybe throw exception if source_plate_uuid is null, because the query {"lh_source_plate_uuid: null} is valid, but we dont want that to happen.

File: tests/helpers/test_mongo.py [hunk 3]
```diff
@@ -81,6 +82,42 @@ def test_get_positive_samples_in_source_plate_returns_none_failure_fetching_samp
...
+                sample.update({FIELD_DATE_TESTED: sample[FIELD_DATE_TESTED].replace(microsecond=0)})
+
+        assert result == expected_samples
+
+
+def test_get_all_samples_for_source_plate_returns_empty_list_no_matching_samples(app, samples):
+    with app.app_context():
+        result = get_all_samples_for_source_plate("source plate uuid does not exist")
+
+        assert result == []
```
Comment (emrojo) [inline tests/helpers/test_mongo.py:108]: We can add a test when the argument is None

File: lighthouse/routes/common/plates.py (hunk 1 above)
Comment (sdjmchattie) [reply lighthouse/routes/common/plates.py:53]: Yes perhaps I should be more explicit, though in Python, any iteration of a dictionary without specifying a method on it just iterates the keys.

File: lighthouse/helpers/mongo.py (hunk 2 above)
Comment (sdjmchattie) [reply lighthouse/helpers/mongo.py:81]: It can't happen because the type hinting explicitly specifies that the value is a string that cannot be `None` but I will add the check because I think this is something that could easily slip up in future.  I can't add a test for it because mypy will complain that I'm passing a `None` value to a method that doesn't accept `None`.

File: tests/helpers/test_mongo.py (hunk 3 above)
Comment (sdjmchattie) [reply tests/helpers/test_mongo.py:116]: I've done this and pushed up the change.  I think we probably should revert that last commit though.  Look at it and tell me what you think.  mypy is already protecting from this issue occurring through the typing system.

File: lighthouse/helpers/mongo.py (hunk 2 above)
Comment (emrojo) [reply lighthouse/helpers/mongo.py:87]: Yes, looks good

**Review by emrojo — APPROVED** [review]
